src/core/options-updater.h

```diff
@@ -0,0 +1,45 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2015 Intel Corporation. All Rights Reserved.
+#pragma once
+
+
+#include <librealsense2/h/rs_option.h>
+#include <src/core/option-interface.h>
+
+#include <rsutils/signal.h>
+#include <rsutils/concurrency/concurrency.h>
+
+#include <map>
+#include <chrono>
+#include <functional>
+#include <memory>
+
+
+namespace librealsense {
+
+
+class options_updater
```

Comment:
Need docs.

Reply:
Added purpose description.

Don't think that `watcher` is applicable here. `watcher` is more passive, here we actively check for value changes. `option_update_notifier`?

Reply:
> `watcher` is more passive, here we actively check for value changes

That's an implementation detail. It's the purpose that's named, not how it's done.
It watches for option changes. Just my opinion.